src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -375,13 +380,10 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
             _initMessage = initEvent.Message.WorkerInitResponse;
             _workerChannelLogger.LogDebug("Worker capabilities: {capabilities}", _initMessage.Capabilities);
 
-            if (_initMessage.WorkerMetadata != null)
-            {
-                _initMessage.UpdateWorkerMetadata(_workerConfig);
-                var workerMetadata = _initMessage.WorkerMetadata.ToString();
-                _metricsLogger.LogEvent(MetricEventNames.WorkerMetadata, functionName: null, workerMetadata);
-                _workerChannelLogger.LogDebug("Worker metadata: {workerMetadata}", workerMetadata);
-            }
+            // In placeholder scenario, the capabilities and worker metadata will not be available
+            // until specialization is done (env reload request). So these can be removed from worker init response code path.
```

Comment:
We should check our understanding here with @soninaren as I remember change in flow on placeholder side for the work - selective loading of extensions.